### PR TITLE
Bl 650 fe refactor coder heroes with a single dashboard

### PR DESCRIPTION
**Pull request by dshipuk**
## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Loom Video
https://www.loom.com/share/418e1d9128aa47feab3dd94dc25dd052

## Type of change

Please delete options that are not relevant.
- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have removed unnecessary comments/console logs from my code
- [x] I have made corresponding changes to the documentation if necessary (optional)
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] No duplicate code left within changed files
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

**Review by techlando — APPROVED** [review]
Clean and well written code. Easy to understand, good job using switch cases. 👍👍

**Review by reneetro — APPROVED** [review]
Much nicer with the single cohesive dashboard!!!

**Review by macrowell70 — APPROVED** [review]
Code looks good, clean and easy to understand. One minor nitpick I have is regarding the if statement in NavBar.js. Checking for firstLogin and role_id seems a little counter intuitive, since if logging in, one should have a role_id. I might use a variable that defaults to false instead (e.g. if (!loggedIn) { history.push('/dashboard'); setLoggedIn(true) }. I like your use of switch cases and all in all it looks really good. Great job!

**Review by Roadlyfe — DISMISSED** [review]
looks good! clean code with no obvious bugs.

**Review by Cassius-Cassity — DISMISSED** [review]
Excellent work!

**Comment by lisamdespain** [comment]
I really like that you used a switch statement to determine which dashboard to load. It seems so much more efficient. Your code is ready for when all those components are build out. It will be easy to make room for them. Good job!

**Review by vasilii-garanin — DISMISSED** [review]
Hi guys, good job with dashboard redirect based on role id. I wish you showed your local deployment in Loom video, but then again maybe that would not show us anything if all home pages are not complete yet. Thanks for walking us through your code.

**Review by AnthonyLayne — DISMISSED** [review]
Looks good, saw a few console.logs in there that could probably be taken out, but other than that, great job.

**Review by tkordysh — DISMISSED** [review]
Good job with this one, code looks good!

**Review by reneetro — APPROVED** [review]
This looks really nice and clean! I like the use of switch cases to generate the appropriate home page 10/10

File: src/components/pages/Dashboard/index.js [hunk 1]
```diff
@@ -1,22 +1,43 @@
...
+      return console.log('Role_ID not Found');
+  }
+};
+
+const mapStateToProps = state => {
+  return { user: state.userReducer };
 };
 
-export default InstructorDashboard;
+export default connect(mapStateToProps)(InstructorDashboard);
```
Comment (paulstgermain) [inline src/components/pages/Dashboard/index.js:43]: This should be exporting `Dashboard.js`, no? 😄

File: src/components/pages/Dashboard/index.js [hunk 2]
```diff
@@ -1,22 +1,43 @@
...
-        <Content>
-          <InstructorStats />
-          {/* Other Dashboard Components */}
-        </Content>
-      </Layout>
-    </div>
-  );
+import { connect, useDispatch } from 'react-redux';
+
+const InstructorDashboard = props => {
```
Comment (paulstgermain) [inline src/components/pages/Dashboard/index.js:11]: We should rename this "Dashboard" to match the folder and import in our `index.js`!

**Review by paulstgermain — CHANGES_REQUESTED** [review]
Smooth implementation here, and some unique problem solving on display!

Just 2 small changes I'd love for you to make before this gets merged up:

File: src/components/pages/Dashboard/index.js (hunk 1 above)
Comment (dshipuk) [reply src/components/pages/Dashboard/index.js]: Alright just pushed these changes

File: src/components/pages/Dashboard/index.js (hunk 2 above)
Comment (dshipuk) [reply src/components/pages/Dashboard/index.js]: Alright just pushed these changes

**Review by Roadlyfe — APPROVED** [review]
looks good, I love the use of the switch statement.

**Review by lisamdespain — APPROVED** [review]
I really like that you used a switch statement to determine which dashboard to load. It seems so much more efficient. Your code is ready for when all those components are build out. It will be easy to make room for them. Good job!

**Review by paulstgermain — APPROVED** [review]

**Review by AlexiusThomas — COMMENTED** [review]
Great job with all the detail and references in this document!

**Pull request merged**